sesman/config.c

```diff
@@ -166,10 +172,46 @@ config_read_globals(int file, struct config_sesman *cf, struct list *param_n,
         cf->enable_user_wm = 0;
     }
 
-    if ('\0' == cf->default_wm[0])
+    if (cf->default_wm == 0)
+    {
+        cf->default_wm = g_strdup("startwm.sh");
+    }
+    else if (g_strlen(cf->default_wm) == 0)
+    {
+        g_free(cf->default_wm);
+        cf->default_wm = g_strdup("startwm.sh");
+    }
+    /* if default_wm doesn't begin with '/', it's a relative path to XRDP_CFG_PATH */
+    if (cf->default_wm[0] != '/')
+    {
+        length = g_strlen(XRDP_CFG_PATH) + g_strlen(g_cfg->default_wm) + 2; /* '/' + '\0' */
```

Comment:
using `strlen` on `XRDP_CFG_PATH` is redundant, you better use `sizeof(XRDP_CFG_PATH)-1` since its already known at compile time.